experiments/kompanion/cmd/migrator/migrator.go

```diff
@@ -0,0 +1,369 @@
+
+	unstructured.RemoveNestedField(r.Object, "status", "conditions")
+	unstructured.RemoveNestedField(r.Object, "status", "creationTimestamp")
+	unstructured.RemoveNestedField(r.Object, "status", "healthy")
+	unstructured.RemoveNestedField(r.Object, "status", "observedGeneration")
+	unstructured.RemoveNestedField(r.Object, "status", "selfLink") // Are we sure?
+	status, present, err := unstructured.NestedMap(r.Object, "status")
+	if err != nil {
+		log.Fatalf("Could not get the status field %v\r", err)
+	}
+	if present && len(status) == 0 {
+		unstructured.RemoveNestedField(r.Object, "status")
+	}
+
+	return r
+}
+
+func RunMigrator(ctx context.Context, opts *MigratorOptions) error {
+	log.Printf("Running kompanion export with kubeconfig: %s", opts.Kubeconfig)
+
+	if err := opts.validateFlags(); err != nil {
+		return err
+	}
+
+	config, err := utils.GetRESTConfig(ctx, opts.Kubeconfig)
+	if err != nil {
+		return fmt.Errorf("error building kubeconfig: %w", err)
+	}
+
+	// We rely more on server-side rate limiting now, so give it a high client-side QPS
+	if config.QPS == 0 {
+		config.QPS = 100
+		config.Burst = 20
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return fmt.Errorf("error creating Kubernetes clientset: %sw", err)
+	}
+
+	dynamicClient, err := dynamic.NewForConfig(config)
+	if err != nil {
+		return fmt.Errorf("error creating dynamic client: %w", err)
+	}
+
+	// use the discovery client to iterate over all api resources
+	discoveryClient := clientset.Discovery()
+	var resources []schema.GroupVersionResource
+	resources, err = utils.GetResources(discoveryClient, resources)
+	if err != nil {
+		return fmt.Errorf("error fetching resources: %w", err)
+	}
+
+	outputChannel := make(chan string, 2*opts.WorkerRoutines)
+
+	namespaces, err := clientset.CoreV1().Namespaces().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("error fetching namespaces: %w", err)
+	}
+	shouldExcludeObject := func(id types.NamespacedName) bool {
+		if shouldExclude(id.Namespace, opts.IgnoreNamespaces, opts.TargetNamespaces) {
+			return true
+		}
+		return shouldExclude(id.Name, opts.IgnoreObjects, opts.TargetObjects)
+	}
+
+	// write the namespaces to the migrator file
+	filedir := filepath.Dir(opts.MigrationFile)
+	if err := os.MkdirAll(filedir, 0o700); err != nil {
+		return fmt.Errorf("error creating directory %s: %w", filedir, err)
+	}
+	migfile, err := os.Create(opts.MigrationFile)
+	if err != nil {
+		return fmt.Errorf("error creating migration file %s: %w", opts.MigrationFile, err)
+	}
+	defer migfile.Close()
+	nsGvr := schema.GroupVersionResource{
+		Group:    "",
+		Version:  "v1",
+		Resource: "namespaces",
+	}
+	nss, err := dynamicClient.Resource(nsGvr).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("fetching namespace %s resources: %w", nsGvr, err)
+	}
+	for _, ns := range nss.Items {
+		ns = pruneResource(ns)
+		if len(ns.GetAnnotations()) <= 0 {
+			continue
+		}
```

Comment:
for my own learning, are we saying we don't wan to migrate namespaces without any annotations ?